src/Runtime/jni/jnilog.c

```diff
@@ -140,21 +218,23 @@ static FILE *get_log_file_by_name(char *name) {
  */
 void log_init() {
 
-  if (log_inited)
+  PTHREAD_ONCE(&key_once, key_init);
```

Comment:
I think it is confusing to have a single PTHREAD_* macro in an otherwise generic function. I think it would be cleaner if the functionality was hidden inside a function that can be a noop on Windows (or other platforms that don't support pthread). Having said that, this is better than using the ifdef, so if you want to leave it this way, that's fine too.

Reply:
I sense you are in general against macros. Since there are no clear rules on when one should use macros and when one should use functions, this mostly comes down to personal coding style. My general rule of thumbs is that if something is relatively simple and can be determined at compile time I would use macros. So unless the code vastly deviates from "common standards", I suggest we don't debate on coding style too much since it quickly becomes fruitless.

Reply:
I don't mind macros, they can be really useful. My comment here was more about the naming - on Windows, we don't really use pthread* but a completely different set of functions, so having the macros be named PTHREAD_ONCE or PTHREAD_SELF, etc. makes the code read as less platform-agnostic. My suggestion would be to name them something that doesn't have pthread in the name since they don't always refer to actual pthread functions/objects.

Reply:
When everyone uses pthread except Windows, it would be bending backwards to not use pthread/PTHREAD in the name.